ql/math/array.hpp

```diff
@@ -516,12 +570,21 @@ namespace QuantLib {
         return result;
     }
 
+    inline Array operator+(Array&& v) {
+        return std::move(v);
```

Comment:
This should be
```suggestion
        return v;
```
to allow the compiler to elide the copy (or move) entirely.

Reply:
If `v` were a local variable, that would be true, but here it's passed as an rvalue reference to the function, so `std::move` is required in order to preserve the rvalue-ness of the variable within the scope of the function. This is discussed in Scott Meyers' _Effective Modern C++_ Item 25.

Some compilers might be able to elide the copy or move even without `std::move` but GCC [does not](https://godbolt.org/z/6eG8ePas4) even with `-O3`.